pkg/contract-manager/cr_services.go

```diff
@@ -0,0 +1,150 @@
+// getAllFlavoursCRs returns all the Flavour CRs in the cluster
+func getAllFlavoursCRs() ([]nodecorev1alpha1.Flavour, error) {
+	cl, err := localResourceManager.GetKClient(context.Background())
+	utilruntime.Must(err)
+
+	var flavourList nodecorev1alpha1.FlavourList
+
+	// List all Flavour CRs
+	err = cl.List(context.Background(), &flavourList)
+	if err != nil {
+		return nil, err
+	}
+
+	var flavours []nodecorev1alpha1.Flavour
+	for _, flavour := range flavourList.Items {
+		fCR := &nodecorev1alpha1.Flavour{
+			Spec: flavour.Spec,
+		}
+		flavours = append(flavours, *fCR)
+	}
+
+	return flavours, nil
+}
+
+// getFlavourByID returns the Flavour CR in the cluster that matches the flavourID
+func getFlavourByID(flavourID string) (*nodecorev1alpha1.FlavourSpec, error) {
+	flavoursSpec, err := getAllFlavoursCRsSpec()
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavour := range flavoursSpec {
+		if flavour.FlavourID == flavourID {
+			return &flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// getFlavourByIDCR returns the entire Flavour CR (not only spec) in the cluster that matches the flavourID
+func getFlavourByIDCR(flavourID string) (*nodecorev1alpha1.Flavour, error) {
+	flavoursSpec, err := getAllFlavoursCRs()
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavour := range flavoursSpec {
+		if flavour.Spec.FlavourID == flavourID {
+			return &flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// getFlavoursBySelector returns the Flavour CRs in the cluster that match the selector
+func getFlavoursBySelector(flavoursSpec []nodecorev1alpha1.FlavourSpec, selector Selector) ([]nodecorev1alpha1.FlavourSpec, error) {
+	var flavours []nodecorev1alpha1.FlavourSpec
+
+	// Get the Flavours that match the selector
+	for _, flavour := range flavoursSpec {
+		cpuInt, _ := flavour.Characteristics.Cpu.AsInt64()
+		ramInt, _ := flavour.Characteristics.Memory.AsInt64()
+		if string(flavour.Type) == selector.FlavourType &&
+			int(cpuInt) == selector.CPU &&
+			int(ramInt) == selector.RAM {
+			flavours = append(flavours, flavour)
+		}
+	}
+
+	return flavours, nil
+}
+
+// createContractCustomResource creates a Contract CR
+func createContractCustomResourceSeller(ctx context.Context, flavour nodecorev1alpha1.Flavour, buyerID string) *reservationv1alpha1.Contract {
+	contract := &reservationv1alpha1.Contract{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "contract-" + flavour.Spec.FlavourID,
+			Namespace: discoverymanager.DEFAULT_NAMESPACE,
+		},
+		Spec: reservationv1alpha1.ContractSpec{
+			Flavour: flavour,
+			BuyerID: buyerID,
+			Seller:  flavour.Spec.Owner,
+		},
+		Status: reservationv1alpha1.ContractStatus{
+			Phase: nodecorev1alpha1.PhaseStatus{
+				Phase: nodecorev1alpha1.PhaseSolved,
+			},
+		},
+	}
+
+	return contract
+}
```

Comment:
I do not understand, a Contract is not a Flavour, since I also asked to Lorenzo: a Contract is not a Flavour but refers to a Flavour. Why is it wrong what i did??

Reply:
AFTER CALL & CONVERSATION ON TELEGRAM: no longer wrap a `Flavour`, just its `FlavourID` (like in the Reservation).